### PR TITLE
Migrate tests to pytest

### LINKED ISSUE #75: Convert tests to pytest
Would you consider converting the tests to pytest?

`assert expr` is easier to read than `self.assertSomething(x, y)`

Using their `tmp_path` fixture would simplify the temporary files manipulation enormously. Every test function gets its own subdirectory in `/tmp/pytest-of-$USER` and three latest test runs are kept there for further investigation. This would save the whole housekeeping after tests.

Reorganizing the tests to `/tests` and the testdata to `/tests/testdata` would also make the ubiquitous `relative_path_to` method redundant.

And, last, but not least, organizing tests to simple methods would remove one indentation level from the test files.

### PR DESCRIPTION
**Pull request by eumiro**
This is a longer branch with incremental migration of tests to pytest.

1. Prepare the infrastructure to use `tests/` as the test directory
2. Migrate the `test_barcodes` test module together with heavily edited `utilities.py` file (and remove the `test/barcodes` directory)

Please, share your ideas/questions/improvements in the discussion here.

It is a draft PR to avoid accidental merge. New commits will appear here later.

When done, closes #75

### CODE REVIEW

**Comment by eumiro** [comment]
Please ignore the `codecov` reports for now. The first commit in this PR deactivates all existing tests and runs only those migrated within this branch.

File: tox.ini [hunk 1]
```diff
@@ -17,9 +17,11 @@ python =
     3.9: py39
 
 [testenv]
+deps =
```
Comment (Lucas-C) [inline tox.ini:20]: Please use `test/requirements.txt` here to avoid duplicating dependencies

**Review by Lucas-C — CHANGES_REQUESTED** [review]

**Comment by Lucas-C** [comment]
Thanks for starting this work.

First off: do you plan to migrate all tests to `pytest` in a single PR or not?

Depending on the answer to this question, my comments below may be inaccurate.

Overall, I'm not really a big fan of the global strategy to migrate to a 2nd directory named `tests`, for several reasons:
- I don't really see the benefit of it: why migrating tests one by one? Given the goal here, the tests logic should not vary in this effort, only the "glue" around executing them.
- it is very important that the tests continue to pass in the `GitHub Actions` pipeline during this effort,
and they are already launched by `pytest`: https://github.com/PyFPDF/fpdf2/blob/master/.github/workflows/continuous-integration-workflow.yml#L39
- having a separate directory makes PRs difficult to read: many files are just moved around "as is", and moved files can be considered removed & recreated by `git` instead of just altered

Hence, my suggestion would be to incrementally add usage of `pytest` features without changing the files structures:
- renaming all test files with a `test_` prefix to be more standard and get rid of [this custom configuration entry](https://github.com/PyFPDF/fpdf2/blob/master/setup.cfg#L18)
- moving from classes to individual test functions
- introducing the use of `tmp_path`
- etc.

**Comment by eumiro** [comment]
@Lucas-C since I am relatively new to this test suite, I first wanted to go class by class in order to see working results fast. Already in the first tiny `test_barcodes` I could refactor the `assert_pdf_equal` and related methods. But now after two migrated modules I see the structure and I could do it the way you want to have it, i.e. one systemic change on the whole test suite at a time per PR with everything working in between.

**Comment by Lucas-C** [comment]
Sounds good 😉

I understand your initial approach, not knowing much about the existing tests.

I already merged https://github.com/PyFPDF/fpdf2/pull/81 you just submitted: thanks!

Feel free to close or rebase this PR to continue.

And if there are tests that prove difficult to adapt to `pytest`, do not hesitate to ask for help or ideas on how to deal with them 
😊

**Comment by eumiro** [comment]
This PR can be closed. Please follow #82

**Pull request closed without merging**